crates/biome_js_semantic/src/events.rs

```diff
@@ -332,19 +327,18 @@ impl SemanticEventExtractor {
         }
     }
 
-    fn is_var(binding: &impl AstNode<Language = JsLanguage>) -> Option<bool> {
-        let declarator = binding.parent::<JsVariableDeclarator>()?;
-        let is_var = match declarator.syntax().parent().kind() {
-            Some(JsSyntaxKind::JS_VARIABLE_DECLARATOR_LIST) => declarator
-                .parent::<JsVariableDeclaratorList>()?
-                .parent::<JsVariableDeclaration>()?
-                .is_var(),
-            Some(JsSyntaxKind::JS_FOR_VARIABLE_DECLARATION) => {
-                declarator.parent::<JsForVariableDeclaration>()?.is_var()
+    fn is_var(declarator: &JsSyntaxNode) -> Option<bool> {
+        debug_assert!(declarator.kind() == JsSyntaxKind::JS_VARIABLE_DECLARATOR);
```

Comment:
nit: let's add a message to this debug assertion. Usually, this text should tell the developer how they could fix the issue, or just a hint.